internal/datastore/crdb/watch.go

```diff
@@ -361,6 +363,47 @@ func (cds *crdbDatastore) processChanges(ctx context.Context, changes pgx.Rows,
 
 			for _, revChange := range filtered {
 				revChange := revChange
+
+				// TODO(jschorr): Change this to a new event type if/when we decide to report these
+				// row GCs.
+
+				// Filter out any DELETEs that occurred due to a relationship being expired and CRDB
+				// performing GC of the row.
+				// As CRDB does not (currently) provide a means for differentiating rows deleted by a user
+				// and rows deleted by the system, we use a set of heuristics to determine when to filter:
+				//
+				// 1) Rows deleted by the TTL cleanup in CRDB will not have any metadata associated with the
+				//    transaction.
+				// 2) No other operations besides DELETE will be performed by the TTL cleanup.
+				// 3) Only filter rows that had an expires_at that are before the current time.
+				if _, ok := revChange.Metadata.AsMap()[spicedbTransactionKey]; !ok {
+					if len(revChange.ChangedDefinitions) == 0 && len(revChange.DeletedNamespaces) == 0 &&
+						len(revChange.DeletedCaveats) == 0 {
+						hasNonTTLEvent := false
+						for _, relChange := range revChange.RelationshipChanges {
```

Comment:
> it would not filter expired relationships if it finds a delete first

You mean if it finds a *non-delete*?

> it would not filter expired relationships if it finds a tuple without expiration first

Which is correct: there cannot, by definition, be a relationship without an expiration in the TTL-based deletion transaction

> it would filter relationships that are not expired, or have no expiration because it found first an expired relationship.

No? It checks all the updates in the change

Reply:
we discussed offline, this is fine:
- this code will only handle transactions originating from GC'ed information, which is only generated by CRDB, never from user-generated transactions
- in CRDB stuff will be GC'ed after 24h, so that `time.Now()` call there drifting is irrelevant